docker/nwchem/Dockerfile

```diff
@@ -0,0 +1,89 @@
+FROM        ubuntu:14.04
```

Comment:
ubuntu 14.04 is at end of life (April 2019), is there a specific reason why we can't use 16.04 / 18.04?

Reply:
NWChem 6.6, the version used in https://github.com/wadejong/NWChem-Json, does not compile with 18.04. With 16.04 applying this patch:

```

diff --git a/docker/nwchem/Dockerfile b/docker/nwchem/Dockerfile
index 6732d93..ef0eaea 100644
--- a/docker/nwchem/Dockerfile
+++ b/docker/nwchem/Dockerfile
@@ -1,4 +1,4 @@
-FROM        ubuntu:14.04
+FROM        ubuntu:16.04
 
 LABEL maintainer="Muammar El khatib <melkhatibr@lbl.gov>, Chris Harris <chris.harris@kitware.com>"
 
@@ -9,22 +9,25 @@ ENV         NWCHEM_TOP="/opt/nwchem"
 RUN         apt-get update \
             && apt-get -y upgrade \
             && apt-get install -y gfortran libopenblas-dev libopenmpi-dev \
-                                  openmpi-bin tcsh make ssh patch curl git
+                                  openmpi-bin tcsh make ssh patch curl git \
+            && apt-get install -y python3 python3-pip
 RUN         apt-get install -y software-properties-common \
-            && add-apt-repository -y ppa:ubuntu-toolchain-r/test \
-            && add-apt-repository ppa:jonathonf/python-3.6 \
+            #&& add-apt-repository -y ppa:ubuntu-toolchain-r/test \
+            #&& add-apt-repository ppa:jonathonf/python-3.6 \
             && apt-get update \
-            && apt-get install -y gfortran-4.9 python3.6 python3.6-dev \
+            && apt-get install -y gfortran-4.9 \
             && update-alternatives --install /usr/bin/gfortran gfortran /usr/bin/gfortran-4.9 60
 
 WORKDIR     /app/
 COPY        requirements.txt /app/
-RUN         curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py \
-            && python3.6 get-pip.py \
-            && rm get-pip.py \
-            && ln -s /usr/bin/python3.6 /usr/local/bin/python3 \
-            #&& ln -s /usr/local/bin/pip3 /usr/local/bin/pip3 \
-            && /usr/local/bin/pip3 install -r /app/requirements.txt
+#RUN         curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py \
+#            && python3.6 get-pip.py \
+#            && rm get-pip.py \
+#            && ln -s /usr/bin/python3.6 /usr/local/bin/python3 \
+#            #&& ln -s /usr/local/bin/pip3 /usr/local/bin/pip3 \
+#            && /usr/local/bin/pip3 install -r /app/requirements.txt
+RUN         pip3 install -r /app/requirements.txt
+
 COPY        src/* /app/
 
 WORKDIR     /opt
diff --git a/docker/nwchem/requirements.txt b/docker/nwchem/requirements.txt
index 3b002f6..6e8822b 100644
--- a/docker/nwchem/requirements.txt
+++ b/docker/nwchem/requirements.txt
@@ -2,4 +2,3 @@ avogadro
 click
 jinja2
 openchemistry
--e git+git://github.com/cclib/cclib.git#egg=cclib

```

makes compilation fail with:

```
checking mpi.h presence... yes
checking for mpi.h... yes
checking for library containing MPI_Init... no
configure: error: test for COMEX_NETWORK=MPI_TS failed
configure: error: ../../ga-5-4/comex/configure failed for comex
make[1]: *** [build/config.status] Error 1
GNUmakefile:551: recipe for target 'build/config.status' failed
GNUmakefile:106: recipe for target 'libraries' failed
make: *** [libraries] Error 1
Removing intermediate container fb95614044a1
 ---> 31e7f17f0d55
Step 33/52 : WORKDIR     ${NWCHEM_TOP}/src/util
 ---> Running in 30855280288a
Removing intermediate container 30855280288a
 ---> ebc68ad4e607
Step 34/52 : RUN         make ; make && cp *.mod ../include
 ---> Running in 0e54e1868fa3
( cd /opt/nwchem/src/config; make /opt/nwchem/bin/LINUX64/depend.x ; )
/usr/bin/gcc  -g -o /opt/nwchem/bin/LINUX64/depend.x depend.c
/bin/rm -f depend.o
/opt/nwchem/bin/LINUX64/depend.x -I/usr/lib/openmpi/include -I/opt/nwchem/src/tools/install/include > dependencies                                                                                                
gfortran -c -m64 -ffast-math  -Warray-bounds -fdefault-integer-8 -march=native -mtune=native -finline-functions -O2 -g -fno-aggressive-loop-optimizations -g -O   output.f                                        
Making include_stamp in md5
make: *** No rule to make target '/opt/nwchem/src/include/mafdecls.fh', needed by '/opt/nwchem/lib/LINUX64/libnwcutil.a(errquit.o)'.  Stop.                                                                       
make: *** Waiting for unfinished jobs....
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
/opt/nwchem/bin/LINUX64/depend.x  -I/opt/nwchem/src/tools/install/include > dependencies
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
cp -p checksum.h /opt/nwchem/src/include
touch include_stamp
cp -p util.fh itri.fh msgids.fh  numerical_constants.fh stdio.fh printlevels.fh bitops.fh bitops_decls.fh bitops_funcs.fh bgj.fh bgj_common.fh nwc_const.fh errquit.fh util_sgroup.fh util_params.fh utilc_nwchem_srcdir.fh /opt/nwchem/src/include
touch include_stamp
gfortran -c -m64 -ffast-math  -Warray-bounds -fdefault-integer-8 -march=native -mtune=native -finline-functions -O2 -g -fno-aggressive-loop-optimizations -g -O   output.f                                        
make: *** No rule to make target '/opt/nwchem/src/include/mafdecls.fh', needed by '/opt/nwchem/lib/LINUX64/libnwcutil.a(errquit.o)'.  Stop.                                                                       
make: *** Waiting for unfinished jobs....
The command '/bin/sh -c make ; make && cp *.mod ../include' returned a non-zero code: 2
```

Reply:
Stick with 14.04